### PR TITLE
Release v0.4.0 (closes #51)

### LINKED ISSUE #51: Release?
OK, I think I have made all of the updates/improvements that I can to bring this closer to current exporter practices. The last tag was v0.3.0 some years ago. Are we ready for a v0.4.0 or v1.0.0?

### PR DESCRIPTION
**Pull request by DaAwesomeP**
I hopefully didn't miss anything in the changelog!

cc @SuperQ

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,23 @@
...
+* [ENHANCEMENT] Corrected various typos in docs, comments, and diagrams
+* [ENHANCEMENT] Cleanup uncessary metrics (#21)
+* [ENHANCEMENT] Improve and clarify example (#26)
+* [ENHANCEMENT] Set metrics_path in prometheus.yml (#31)
+* [ENHANCEMENT] Remove stale license information(#33)
+* [ENHANCEMENT] README: mention that sub_module is a required parameter and target needs to include the port (#36)
+* [ENHANCEMENT] Cleanup build (#43)
+* [ENHANCEMENT] Fix README Diagram (#45)
+* [ENHANCEMENT] Fix another diagram typo (#48)
+* [ENHANCEMENT] Fix CodeQL Error (#50)
```
Comment (SuperQ) [inline CHANGELOG.md:17]: We typically only list user visible changes. We don't need to include stuff like test fixes or build stuff.

Comment (DaAwesomeP) [reply CHANGELOG.md]: Fixed!

**Review by SuperQ — APPROVED** [review]

**Comment by DaAwesomeP** [comment]
Ah it seems `secrets.RICHIH_GITHUB_TOKEN` was not set so the artifacts did not upload: https://github.com/RichiH/modbus_exporter/actions/runs/4409395100/jobs/7725704339#step:4:564

**Comment by RichiH** [comment]
Can you link me to the docs/a howto on what I should do? I looked at it with @bastischubert during a call yesterday and we didn't find it.

**Comment by RichiH** [comment]
https://github.com/RichiH/modbus_exporter/pull/53

**Comment by SuperQ** [comment]
Release works now: https://github.com/RichiH/modbus_exporter/releases/tag/v0.4.1

**Pull request merged**